src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -85,5 +104,95 @@ where kvp.Key.Trim().StartsWith("dotnet_naming_rule.", StringComparison.Ordinal)
                 where nameSplit.Length == 3
                 select nameSplit[1])
                 .Distinct();
+
+        private abstract class NamingRuleSubsetComparer : IComparer<NamingRule>
+        {
+            protected NamingRuleSubsetComparer()
+            {
+            }
+
+            public int Compare(NamingRule x, NamingRule y)
+            {
+                var firstIsSubset = FirstIsSubset(in x, in y);
+                var secondIsSubset = FirstIsSubset(in y, in x);
+                if (firstIsSubset)
+                {
+                    return secondIsSubset ? 0 : -1;
+                }
+                else
+                {
+                    return secondIsSubset ? 1 : 0;
+                }
+            }
+
+            protected abstract bool FirstIsSubset(in NamingRule x, in NamingRule y);
```

Comment:
May be worth a comment that non-proper subsets should return true. (Yes, it's the mathematical definition but assuming all developers are mathematicians isn't great.)